### PR TITLE
feat(circuits): add a noir circuit for merkle tree membership and JS tests for poseidon2 and pedersen

### PR DESCRIPTION
**Pull request by signorecello**
Follow-up to #234, this PR:

- Adds a merkle tree membership circuit in Noir. 
- Adds an abstraction layer to both MTs to make it easier to simply call for example `mt.add` to prove addition. Sweet Noir traits!
- Adds a bunch of noir tests
- Refactors the LeanIMT tests just like I did for #234, featuring `poseidon`, `poseidon2` and `pedersen` hashes.
- Abstracts tree sizes away by making the libs use slices instead. This makes them useable with any array size.
- Merges all the Merkle Trees implementations and tests into one single lib. This makes most of the code reusable, and makes the interface look similar.
- Adds READMEs with instructions

### CODE REVIEW

**Review by vplasencia — APPROVED** [review]
Great work. Thank you very much.

**Comment by signorecello** [comment]
Seems like I cannot merge, can you do it please @vplasencia ?

**Comment by vplasencia** [comment]
Hey @signorecello! At least two people should review the PR before merging. When another of the reviewers reviews it, we will merge it. Thanks.

**Review by cedoor — APPROVED** [review]

**Pull request merged**
